Replace proxy by forward (#312)
giantswarm/docs: src/content/guides/advanced-coredns-configuration/index.md

The cache plugin also supports much more detailed configuration which is documented in the [upstream documentation](https://coredns.io/plugins/cache/).

> ## Additional forwards (formerly known as proxy)

Reviewer: ```suggestion
## Additional forwards (formerly known as proxy) {#additional-forwards}
```

This will make the ID for this headline `additional-forwards`, which allows for some longer living link targets once we change that headline again.

The cache plugin also supports much more detailed configuration which is documented in the [upstream documentation](https://coredns.io/plugins/cache/).

## Additional forwards (formerly known as proxy) {#additional-forwards}

> In CoreDNS version `1.4.0` the proxy plugin has been deprecated. The same behaviour can be achieved now with forward although syntax can be a bit different. The forward plugin indeed has better performance because it reuses opened upstream connections.

Reviewer 2: ```suggestion
In CoreDNS version `1.4.0` the proxy plugin has been deprecated. The same behaviour can be achieved now with forward although the syntax can be a bit different. The forward plugin has better performance because it reuses opened upstream connections.
```

The default forward entry we set in CoreDNS is

```yaml
forward . /etc/resolv.conf
```

> You can add additional forward entries by adding a each as a line to the foward field of the user ConfigMap. They will be select by sequential order.

Reviewer 2: ```suggestion
You can add additional forward entries by adding a each as a line to the forward field of the user ConfigMap. They will be selected in sequential order.
```

You can use a simple line or multiple lines to define the upstreams of the default server block.

```yaml
data:

>   foward: 1.1.1.1

Reviewer 2: ```suggestion
  forward: 1.1.1.1
```

```

```yaml
data:

>   foward: |

Reviewer 2: ```suggestion
  forward: |
```

    1.1.1.1
    8.8.8.8
```

__Warning:__ The number of forward upstreams is limited to 15.

> Above example would result in following additional forward entries in the CoreDNS configuration:

Reviewer 2: ```suggestion
Above example would result in the following additional forward entries in the CoreDNS configuration:
```

```yaml
forward . 1.1.1.1 /etc/resolv.conf
```

```yaml
forward . 1.1.1.1 8.8.8.8 /etc/resolv.conf
```

This setting would forward all requests to 1.1.1.1 which is Cloudflare's DNS. If the first upstream fails the second IP (8.8.8.8) will be used as resolver. In case it fails too, all requests will be resolved by the default DNS provider set for your cluster.

The forward plugin also supports much more detailed configuration which is documented in the [upstream documentation](https://coredns.io/plugins/forward/).

## Advanced configuration

> In case you need to have a finer granularity you can define custom server blocks with all configuration desired. They will be parsed after the catch-all block in the Corefile. As an example, let's define a block for a `example.com` with some custom configuration:

Reviewer 2: ```suggestion
In case you need to have a finer granularity you can define custom server blocks with all desired configuration. They will be parsed after the catch-all block in the Corefile. As an example, let's define a block for a `example.com` with some custom configuration:
```
